wandb/sdk/artifacts/artifact.py

```diff
@@ -1336,7 +1338,8 @@ def do_write(f: IO) -> None:
         # It will be added again later on finalize, but succeed since
         # the checksum should match
         entry = self.add_file(file_path, name, is_tmp_name)
-        self._added_objs[obj_id] = entry
+        # We store a reference to the obj so that its id doesn't get reused.
+        self._added_objs[obj_id] = (obj, entry)
```

Comment:
interesting, is it only our tests that need access to the artifactManfestEntry or does our production code also use it? We would've seen the GC error earlier if prod also relied on it right?

Reply:
this does make me think we are going to hold on to more objects than are needed. I'm thinking about a case where we have a big table of wandb images and we want to garbage collect all those in-memory arrays. 

As a potential alternative, perhaps the WBValue class could retain some small state to reduce the need to keep these object around. One idea: WBValue has a property `runtime_id` which is a random id generated at construction time and that can be used rather than the potentially reused python id.

Reply:
We stored a reference to the WBValue object before my refactoring, see [here](https://github.com/wandb/wandb/blob/ac2ec37ad76a2ccaa5f0d9646fc0682cef4b1ab9/wandb/sdk/artifacts/local_artifact.py#L535). Let's go back to that first, and then we can optimize it in a separate PR.

Reply:
ah gotcha, if this is reverting to previous behavior, ill stamp to unblock